writer.go

```diff
@@ -153,6 +157,10 @@ func (w *Writer) WriteRow(row Row) error { return w.writer.WriteRow(row) }
 // The content of the row group is flushed to the writer; after the method
 // returns successfully, the row group will be empty and in ready to be reused.
 func (w *Writer) WriteRowGroup(rowGroup RowGroup) (int64, error) {
+	if rowGroup.NumRows() == 0 {
+		return 0, nil
+	}
```

Comment:
The row group may have no rows, but it may carry a schema and a list of sorting columns that the programs would intent to write to the file. The check here would prevent this.

Maybe we could add a tests to verify these conditions are supported?